### PR TITLE
Bug 1740857: Backport grpc sourcemanager to 4.1

### PR DESCRIPTION
**Pull request by ecordell**
**Description of the change:**
This backports the grpc sourcemanager to 4.1

**Motivation for the change:**

The sourcemanager will fix an issue in some 4.1 clusters which see high memory usage. 

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage 
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/docs` 
- [ ] Commit messages sensible and descriptive

### CODE REVIEW

File: pkg/controller/operators/catalog/operator.go [hunk 1]
```diff
@@ -211,6 +211,7 @@ func NewOperator(kubeconfigPath string, logger *logrus.Logger, wakeupInterval ti
 		op.lister.CoreV1().RegisterPodLister(namespace, podInformer.Lister())
 		op.lister.CoreV1().RegisterConfigMapLister(namespace, configMapInformer.Lister())
 	}
+	op.sources = grpc.NewSourceStore(logger, 10*time.Second, 10*time.Minute, op.syncSourceState)
```
Comment (jpeeler) [inline pkg/controller/operators/catalog/operator.go]: I don't think this line is necessary, is it?

Comment (ecordell) [reply pkg/controller/operators/catalog/operator.go]: Oh, that's weird! I will fix

**Pull request merged**